### PR TITLE
Tests: use classes instead of paths in settings

### PR DESCRIPTION
**Pull request by elacuesta**
Simplify tests after #3873

### CODE REVIEW

File: tests/test_spidermiddleware_output_chain.py [hunk 1]
```diff
@@ -56,9 +58,9 @@ class ProcessSpiderInputSpiderWithoutErrback(Spider):
     custom_settings = {
         'SPIDER_MIDDLEWARES': {
             # spider
-            __name__ + '.LogExceptionMiddleware': 10,
-            __name__ + '.FailProcessSpiderInputMiddleware': 8,
-            __name__ + '.LogExceptionMiddleware': 6,
```
Comment (wRAR) [inline tests/test_spidermiddleware_output_chain.py:61]: So it looks like it currently uses only one of these entries? :)

**Review by wRAR — APPROVED** [review]
I like this so much more!

File: tests/test_spidermiddleware_output_chain.py (hunk 1 above)
Comment (elacuesta) [reply tests/test_spidermiddleware_output_chain.py:61]: Indeed, the behaviour makes sense. The failing middleware fails in `process_spider_input` and does not define a `process_spider_exception` method, so `LogExceptionMiddleware.process_spider_exception` (the method that logs the exception) is placed at the beginning of the chain as expected. In fact, we only need one of the occurrences for the test case, I'm removing the duplicate one.
Nice catch, thanks for noticing!

Comment (elacuesta) [reply tests/test_spidermiddleware_output_chain.py:61]: (For reference)
current `master` (5a386393):
```
  Enabled spider middlewares:
['tests.test_spidermiddleware_output_chain.LogExceptionMiddleware',
 'tests.test_spidermiddleware_output_chain.FailProcessSpiderInputMiddleware',
 'scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
 ```

current `test-classes-settings` (62394c2b):
```
   Enabled spider middlewares:
[<class 'tests.test_spidermiddleware_output_chain.LogExceptionMiddleware'>,
 <class 'tests.test_spidermiddleware_output_chain.FailProcessSpiderInputMiddleware'>,
 'scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']

**Review by Gallaecio — APPROVED** [review]
